src/LinAlg/SparseMatrix.C

```diff
@@ -1385,11 +1390,12 @@ void SparseMatrix::calcCSR(IntVec& IA, IntVec& JA,
   IA.resize(nrow+1,nnz);
   JA.resize(nnz);
 
-  IA[0] = 0;
+  IA.front() = 0;
   size_t cur_row = 1, ix = 0;
-  for (auto it = elem.begin(); it != elem.end(); ++it, ix++) {
-    JA[ix] = it->first.second-1;
-    while (it->first.first > cur_row)
+  for (const ValueMap::value_type& it : elem)
```

Comment:
it should imo be reserved for iterators.

Reply:
Agree, changed to val instead